### PR TITLE
APEXCORE-502 Unnecessary byte array copy in DefaultKryoStreamCodec.toByteArray

### PR DESCRIPTION
**Pull request by vrozov**
@tweise Please review

### CODE REVIEW

File: engine/src/main/java/com/datatorrent/stram/plan/logical/DefaultKryoStreamCodec.java [hunk 1]
```diff
@@ -56,37 +51,29 @@ public DefaultKryoStreamCodec()
...
 
   @Override
-  public Slice toByteArray(T info)
+  public Slice toByteArray(T o)
   {
-    Slice slice = null;
+    final Output output = new Output(32, -1);
     try {
-      ByteArrayOutputStream os = new ByteArrayOutputStream();
-      Output output = new Output(os);
```
Comment (pramodin) [inline engine/src/main/java/com/datatorrent/stram/plan/logical/DefaultKryoStreamCodec.java:76]: Looks like it is a pass through to the output stream? Granted the extra wrapping is not needed.

Comment (vrozov) [inline engine/src/main/java/com/datatorrent/stram/plan/logical/DefaultKryoStreamCodec.java:76]: Yes, Output already has byte array for storing serialized object, flushing internal byte array to optional OutputStream when OutputStream is ByteArrayOutputStream() is unnecessary and must be avoided.

Comment (pramodin) [inline engine/src/main/java/com/datatorrent/stram/plan/logical/DefaultKryoStreamCodec.java:76]: I see so it is not a pass through when the stream is supplied

Comment (vrozov) [inline engine/src/main/java/com/datatorrent/stram/plan/logical/DefaultKryoStreamCodec.java:76]: It is not a direct pass through. Output creates an internal byte array buffer and flushes it to the OutputStream (if one is provided) when the internal buffer becomes full. As in this case OutputStream is ByteArrayOutputStream anyway, this leads to 3 unnecessary byte arrays copy operations.

**Pull request merged**